librubyfmt/src/render_targets.rs

```diff
@@ -47,6 +89,30 @@ impl BreakableEntry {
         tokens
     }
 
+    fn last_token_is_a_newline(&self) -> bool {
+        match self.tokens.last() {
+            Some(x) => x.is_newline(),
+            _ => false,
+        }
+    }
+
+    fn index_of_prev_hard_newline(&self) -> Option<usize> {
+        self.tokens
+            .iter()
+            .rposition(|v| v.is_newline() || v.is_comment())
+    }
+}
+
+impl BreakableEntry {
+    pub fn new(spaces: ColNumber, delims: BreakableDelims) -> Self {
+        BreakableEntry {
+            spaces,
+            tokens: vec![],
```

Comment:
Seeing `vec![]` next to `HashSet::new()` feels weird to me

```suggestion
            tokens: Vec::new(),
```

Reply:
applied manually